src/Symfony/Component/ErrorHandler/ExceptionHandler.php

```diff
@@ -0,0 +1,187 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\ErrorHandler;
+
+use Symfony\Component\ErrorHandler\Exception\OutOfMemoryException;
+use Symfony\Component\ErrorRenderer\ErrorRenderer\HtmlErrorRenderer;
+use Symfony\Component\ErrorRenderer\Exception\FlattenException;
+use Symfony\Component\HttpKernel\Debug\FileLinkFormatter;
+
+/**
+ * ExceptionHandler converts an exception to a Response object.
+ *
+ * It is mostly useful in debug mode to replace the default PHP/XDebug
+ * output with something prettier and more useful.
+ *
+ * As this class is mainly used during Kernel boot, where nothing is yet
+ * available, the Response content is always HTML.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ * @author Nicolas Grekas <p@tchwork.com>
+ *
+ * @final since Symfony 4.3
+ */
+class ExceptionHandler
+{
+    private $debug;
+    private $charset;
+    private $handler;
+    private $caughtBuffer;
+    private $caughtLength;
+    private $fileLinkFormat;
+    private $htmlErrorRenderer;
+
+    public function __construct(bool $debug = true, string $charset = null, $fileLinkFormat = null)
+    {
+        $this->debug = $debug;
+        $this->charset = $charset ?: ini_get('default_charset') ?: 'UTF-8';
+        $this->fileLinkFormat = $fileLinkFormat;
+    }
+
+    /**
+     * Registers the exception handler.
+     *
+     * @param bool        $debug          Enable/disable debug mode, where the stack trace is displayed
+     * @param string|null $charset        The charset used by exception messages
+     * @param string|null $fileLinkFormat The IDE link template
+     *
+     * @return static
+     */
+    public static function register($debug = true, $charset = null, $fileLinkFormat = null)
```

Comment:
missing types in all of the new ErrorHandler component

Reply:
I gonna make those changes in another PR with that goal, where we can see the diff clearer, now that would complicate the review unnecessary.

Reply:
I agree with yceruto. Adding the ErrorHandler component is a big PR. So we should take care type-hints in an another one.